### PR TITLE
CP-4: Display tabs

### PR DESCRIPTION
**Pull request by a-gvo**
Rebased on top of [misc](https://github.com/phenotips/care-pathways/pull/2).

### CODE REVIEW

**Comment by sdumitriu** [comment]
I'd like to see the tab feature part of core PhenoTips. This is the second extension that needs to add tabs, and we want to move patient matches in a tab very soon.

**Comment by sdumitriu** [comment]
Add three new UIX extension points, one for generic `org.phenotips.entitySheet.tab`, two for `org.phenotips.patientSheet.tab` and `org.phenotips.familySheet.tab`. Don't render the HTML content, like we do for the other sheets, but use custom parameters for the tab label, href, and optional custom css class.

**Comment by a-gvo** [comment]
Rebased on top of [misc](https://github.com/phenotips/care-pathways/pull/2) and depends on [PT-3518](https://github.com/phenotips/phenotips/pull/2506).

File: care-pathways-data/ui/src/main/resources/PhenoTips/CarePathwaysConclusions.xml [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+##
+#set($obj = $doc.getObject('PhenoTips.CarePathwaysConclusions', true))
+#set($class = $obj.xWikiClass)
+#set($discard = $doc.use($obj))
+## Disable the Preview button
+#set($previewenabled = false)
+{{/velocity}}
+
+{{velocity}}
+(% class="" id="care-pathways-conclusions-container"%)(((##
```
Comment (sdumitriu) [inline care-pathways-data/ui/src/main/resources/PhenoTips/CarePathwaysConclusions.xml]: If the class is empty, just don't mention it at all.

File: care-pathways-data/ui/src/main/resources/PhenoTips/CarePathwaysConclusions.xml [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see http://www.gnu.org/licenses/
+-->
+
+<xwikidoc version="1.1">
+  <web>PhenoTips</web>
+  <name>CarePathwaysConclusions</name>
```
Comment (sdumitriu) [inline care-pathways-data/ui/src/main/resources/PhenoTips/CarePathwaysConclusions.xml]: Is this still needed?

File: widgets/ui/src/main/resources/PhenoTips/CarePathways.xml [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+      </use>
+    </class>
+    <property>
+      <cache>long</cache>
+    </property>
+    <property>
+      <code>#template('colorThemeInit.vm')
+
+/**********************************************************************************************
+ * Tabs css.
```
Comment (sdumitriu) [inline widgets/ui/src/main/resources/PhenoTips/CarePathways.xml]: Why is this needed? Should it go in PhenoTips core instead?

File: care-pathways-data/pom.xml [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see http://www.gnu.org/licenses/
+-->
+
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
+    <modelVersion>4.0.0</modelVersion>
```
Comment (sdumitriu) [inline care-pathways-data/pom.xml]: Use 2 spaces for indentation instead of 4.

Comment (a-gvo) [reply care-pathways-data/pom.xml]: Fixed.

File: widgets/ui/src/main/resources/PhenoTips/CarePathways.xml (hunk 3 above)
Comment (a-gvo) [reply widgets/ui/src/main/resources/PhenoTips/CarePathways.xml]: Forgot to update a couple PRs, looks like. Fixed.

File: care-pathways-data/ui/src/main/resources/PhenoTips/CarePathwaysConclusions.xml (hunk 2 above)
Comment (a-gvo) [reply care-pathways-data/ui/src/main/resources/PhenoTips/CarePathwaysConclusions.xml]: Fixed.

File: care-pathways-data/ui/src/main/resources/PhenoTips/CarePathwaysConclusions.xml (hunk 1 above)
Comment (a-gvo) [reply care-pathways-data/ui/src/main/resources/PhenoTips/CarePathwaysConclusions.xml]: Fixed.

Comment (sdumitriu) [reply care-pathways-data/ui/src/main/resources/PhenoTips/CarePathwaysConclusions.xml]: You didn't fix in the other 2 files...
